src/main/java/forestry/arboriculture/worldgen/TreeDecorator.java

```diff
@@ -46,16 +46,16 @@
 public class TreeDecorator {
 	private static final List<IAlleleTreeSpecies> SPECIES = new ArrayList<>();
 	private static final Map<ResourceLocation, Set<ITree>> biomeCache = new HashMap<>();
-	
+
 	@SubscribeEvent
 	public void decorateTrees(Decorate event) {
-		if(event.getType() == Decorate.EventType.TREE){
+		if (event.getType() == Decorate.EventType.TREE) {
 			decorateTrees(event.getWorld(), event.getRand(), event.getPos().getX() + 8, event.getPos().getZ() + 8);
 		}
 	}
 
 	public static void decorateTrees(World world, Random rand, int worldX, int worldZ) {
-		if(!Config.isValidTreeDim(world.provider.getDimension())) {
+		if (!Config.isValidTreeDim(world.provider.getDimension())) {
```

Comment:
If we add `|| Config.generateTreesAmount == 0` here it can efficiently skip decorating trees